File: PF/Cours/00/somme.hs
module Somme where

somme n = sum [1..n]

somme' n = sum (enumFromTo 1 n)

sommePairs n = sum [0,2..n]

sommeImpairs n = sum [1,3..n]

sommeCarres n = sum (map (^2) [1..n])

File: PF/Cours/00/somme.c
int somme(int n) {
    int i, somme = 0;

    for (i = 0; i < n; i++) {
        somme += i;
    }

    return 0;
}

int sommePairs(int n) {
    int i, somme = 0;

    for (i = 0; i < n; i+=2) {
        somme += i;
    }

    return 0;
}

int sommeImpairs(int n) {
    int i, somme = 0;

    for (i = 1; i < n; i+=2) {
        somme += i;
    }

    return 0;
}

int sommeCarres(int n) {
    int i, somme = 0;

    for (i = 0; i < n; i++) {
        somme += i * i;
    }

    return 0;
}
